recordedit/form.controller.js

```diff
@@ -382,10 +362,24 @@
 
             if (isDisabled(column)) return;
 
-            var params = {};
+            var originalTuple,
+                editOrCopy = true,
+                params = {};
 
             // pass the reference as a param for the modal
-            params.reference = column.reference.contextualize.compactSelect;
+            // call to page with tuple to get proper reference
+            if (vm.editMode) {
+                originalTuple = $rootScope.tuples[rowIndex];
+            }else if (context.queryParams.copy) {
+                originalTuple = $rootScope.tuples[0];
+            } else {
+                originalTuple = null;
+                editOrCopy = false;
+            }
+
+            var submissionRow = populateSubmissionRow(vm.recordEditModel.rows[rowIndex], vm.recordEditModel.submissionRows[rowIndex], originalTuple, $rootScope.reference.columns, editOrCopy);
+
+            params.reference = column.reference.filteredRef(submissionRow).contextualize.compactSelect;
```

Comment:
If `filteredRef` is on each `ReferenceColumn`, should it be `column.filteredRef(submissionRow).contextualize.compactSelect`?